libqtile/widget/wlan.py

```diff
@@ -71,8 +74,21 @@ def poll(self):
             essid, quality = get_status(self.interface)
             disconnected = essid is None
             if disconnected:
-                return self.disconnected_message
-
+                if self.use_ethernet:
+                    try:
+                        with open(f"/sys/class/net/{self.ethernet_interface}/operstate", "r") as statfile:
+                            if (statfile.read().strip() == 'up'):
+                                return self.ethernet_message
+                            else:
+                                return self.disconnected_message
+                    except FileNotFoundError:
+                        logger.error(
+                            "%s: Ethernet interface has not been found!",
+                            self.__class__.__name__,
+                        )
```

Comment:
Two things:
1) You don't need `self.__class__.__name__`. The log includes the name of the file already.
2) Can you include some sort of flag so this message doesn't get logged every time the widget polls?

Reply:
Sure I can try to get this worked out tomorrow. Also are the other changes requested from some weeks ago okay? I have had no notice of that yet.

Reply:
You mean my comments? You fixed them nicely, thanks!

Reply:
Oh okay, I was a bit confused since Github still says youre requesting changes

Reply:
Sorry! Approved now

Reply:
Appreciated! I will try to get the other changes done once I have some time